browser/ui/webui/settings/brave_tor_handler.cc

```diff
@@ -390,22 +390,27 @@ void BraveTorHandler::IsTorEnabled(const base::Value::List& args) {
   CHECK_EQ(args.size(), 1U);
   AllowJavascript();
   ResolveJavascriptCallback(
-      args[0], base::Value(!TorProfileServiceFactory::IsTorDisabled()));
+      args[0], base::Value(!TorProfileServiceFactory::IsTorDisabled(
+                   web_ui()->GetWebContents()->GetBrowserContext())));
 }
 
 void BraveTorHandler::OnTorEnabledChanged() {
   if (IsJavascriptAllowed()) {
     FireWebUIListener("tor-enabled-changed",
-                      base::Value(!TorProfileServiceFactory::IsTorDisabled()));
+                      base::Value(!TorProfileServiceFactory::IsTorDisabled(
+                          web_ui()->GetWebContents()->GetBrowserContext())));
   }
 }
 
 void BraveTorHandler::IsTorManaged(const base::Value::List& args) {
   CHECK_EQ(args.size(), 1U);
 
+  const bool is_available = TorProfileServiceFactory::IsTorAvailable(
```

Comment:
let's remove `IsTorAvailable` and add `IsTorManaged` instead.

right now `IsTorAvailable` and `IsTorDisabled` contradict each other.